xonsh/history.py

```diff
@@ -515,20 +514,21 @@ def _show(ns=None, hist=None, start_index=None, end_index=None,
         commands = commands[idx]
 
     if ns and ns.reverse:
-        commands = reversed(commands)
-
-    digits = len(str(max([i for c, t, i in commands])))
-    if alias:
-        for c, t, i in commands:
-            for line_ind, line in enumerate(c.split('\n')):
-                if line_ind == 0:
-                    print('{:>{width}}: {}'.format(i, line,
-                                                   width=digits + 1))
-                else:
-                    print(' {:>>{width}} {}'.format('', line,
-                                                    width=digits + 1))
-    else:
-        return commands
+        commands = list(reversed(commands))
```

Comment:
Test was failing because reversed returns iterator and was being exhausted by the newly fixed digits length test.